### ISSUE TITLE
feat: allow return null from React components

### ISSUE THREAD

**Issue by AndersDJohnson** [nene/eslint-plugin-no-null#2]
We could consider adding an option that would allow returning null from a React component (
- e.g., the render or other helper methods of a class that extends React.Component, or a stateless functional component (a function in PascalCase) - or maybe just any function that uses JSX inside - or even from any file that imports React - whatever is feasible.

**Comment by nene** [comment]
Yeah, that would be a nice addition. Though this detection of is-it-react-component is pretty tricky to do.

Then again, a better solution might be to actually avoid `null` even in react components. For example, instead of returning null, one could do:

```js
const MyComp = () => {
  return <></>;
};
```

**Comment by theamazingfedex** [comment]
My team got around this issue by creating a <Nothing/> component and returning that instead of `null` from the component. The <Nothing/> component just returns null, but has `/* eslint-disable-next-line no-null/no-null */` on it.

Example:
``` js
/* eslint-disable-next-line no-null/no-null */
const Nothing = () => null

const MyComp = () => {
  if (returnNullCondition) return <Nothing/>
  return <>My Component Here</>
}
```

**Comment by nene** [comment]
With goal of keeping this plugin plain and simple, I'm rejecting this feature request.
